### PR TITLE
Bulk prices

### PR DESCRIPTION
**Pull request by Adam-**

### CODE REVIEW

File: http-service/src/main/java/net/runelite/http/service/item/ItemService.java [hunk 1]
```diff
@@ -475,4 +460,30 @@ public void check()
...
+	@Scheduled(fixedDelay = 20_000)
+	public void crawlPrices()
+	{
+		if (tradeableItems == null || tradeableItems.length == 0)
+		{
+			return;
+		}
+
+		int idx = random.nextInt(tradeableItems.length);
+		ItemDefinition current = tradeableItems[idx];
```
Comment (Abextm) [inline http-service/src/main/java/net/runelite/http/service/item/ItemService.java]: There must be a better way to do this

File: http-service/src/main/java/net/runelite/http/service/item/ItemService.java [hunk 2]
```diff
@@ -475,4 +460,30 @@ public void check()
...
+
+		fetchPrice(current.id);
+	}
+
+	@Scheduled(fixedDelay = 1_8000_000) // 30 minutes
+	public void reloadItems() throws IOException
+	{
+		List<ItemDefinition> items = cacheService.getItems();
+		tradeableItems = items.stream()
+			.filter(item -> item.isTradeable)
```
Comment (Abextm) [inline http-service/src/main/java/net/runelite/http/service/item/ItemService.java]: There are some items that are tradable, but not on the GE (burnt fish comes to mind)

Comment (Adam-) [reply http-service/src/main/java/net/runelite/http/service/item/ItemService.java]: I think that our isTradeable is actually isGeTradeable

File: http-service/src/main/java/net/runelite/http/service/item/ItemService.java (hunk 1 above)
Comment (Adam-) [reply http-service/src/main/java/net/runelite/http/service/item/ItemService.java]: suggestions?

File: http-service/src/main/java/net/runelite/http/service/item/ItemController.java [hunk 3]
```diff
@@ -212,4 +222,12 @@ public SearchResult search(@RequestParam String query)
 			})
 			.toArray(ItemPrice[]::new);
 	}
+
+	@RequestMapping("/prices")
+	public ResponseEntity<ItemPrice[]> prices()
+	{
+		return ResponseEntity.ok()
+			.cacheControl(CacheControl.maxAge(30, TimeUnit.MINUTES).cachePublic())
```
Comment (deathbeam) [inline http-service/src/main/java/net/runelite/http/service/item/ItemController.java]: Do you need both cacheControl and manual caching?

**Pull request merged**